### PR TITLE
Allow custom styles for modal dismiss icon

### PR DESCRIPTION
**Pull request by amooabeebadesina**
## Description
This change allows to add custom styles (colours, position etc) for the close button in the modal.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style and guidelines of this project. 
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have updated the **CHANGELOG** document.
- [ ] I have added tests to cover my changes.
- [x] I have performed a self-review of my own code
- [x] I have tested my changes/additions in the latest Chrome, Firefox, and Safari.
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] All new and existing tests passed.
- [ ] I have performed the accessibility audit of my UI changes according to the accessibility doc. 
- [ ] [Buffer Engineers] Someone from the Design team reviewed and approved my changes
- [ ] [Buffer Engineers] I have notified the BDS team of my changes in the #proj-design-system Slack channel

### CODE REVIEW

**Review by guldenmw — APPROVED** [review]
I think it's fine to move forward with this because there currently isn't a better way to handle it, but it's a really good example of why we need a better way to send custom styles via props. This instance doesn't warrant an entirely new component, so there should be an easy way to customize it. 
I don't think the PR needs to be blocked by the discussion, but @federicoweber I'd be curious to hear your thoughts here. I think finding a way to avoid updates like these will be beneficial as they end up complicating the component APIs more.

**Comment by amooabeebadesina** [comment]
Can't attach media when editing PR description, so I'll add context here.

## Context

The current close button in the Modal is wrongly positioned except for Carousel Modals. The behaviour is as shown below

composer_modal

This PR allows to pass a `customStyles`  property to the closeButton prop for modals. In addition to be able to dictate the positioning of the icon, the aim is to also be able to pass custom hover effects, colours etc.

**Review by federicoweber — CHANGES_REQUESTED** [review]
Hey @amooabeebadesina, thanks for working on this one.
I'm sorry to be a blocker, but I'm really hesitant to merge this in.

I think we shouldn't need a custom style to solve this, the close button alignment should be guaranteed in all cases. Given this is not the case, I would consider this a bug, and should be solved within the component without introducing new APIs. Also as @guldenmw mentioned passing style in this way can really complicate the components.

One last thing we are trying to move away from this `Modal` in favor of using the `SimpleModal` ([ref.](https://bufferapp.github.io/ui/#/ui/ui/simplemodal)), so if feasible I will suggest using that one🙂.

**Comment by amooabeebadesina** [comment]
> Hey @amooabeebadesina, thanks for working on this one. I'm sorry to be a blocker, but I'm really hesitant to merge this in.
> 
> I think we shouldn't need a custom style to solve this, the close button alignment should be guaranteed in all cases. Given this is not the case, I would consider this a bug, and should be solved within the component without introducing new APIs. Also as @guldenmw mentioned passing style in this way can really complicate the components.
> 
> One last thing we are trying to move away from this `Modal` in favor of using the `SimpleModal` ([ref.](https://bufferapp.github.io/ui/#/ui/ui/simplemodal)), so if feasible I will suggest using that one🙂.

Thanks for the quick chat @federicoweber. I just requested a review

File: src/documentation/markdown/GettingStarted/CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,8 @@
 # Changelog
 
+## [Unreleased] - 2022-05-23
+- Added customStyles prop to Modal close button
```
Comment (federicoweber) [inline src/documentation/markdown/GettingStarted/CHANGELOG.md:4]: This doesn't apply anymore 😉

**Review by federicoweber — APPROVED** [review]
Thank you for the quick work on this @amooabeebadesina, left a small comment on the CHANGELOG, but everything else looks solid 🙌 🌮

**Pull request merged**